script.js

```diff
@@ -1 +1,318 @@
+};
+
+const getRandomIndex = (max) => Math.floor(Math.random() * max);
+
+const shuffleCards = (cards) => {
+  // Loop over the card deck array once
+  for (let currentIndex = 0; currentIndex < cards.length; currentIndex += 1) {
+    // Select a random index in the deck
+    const randomIndex = getRandomIndex(cards.length);
+    // Select the card that corresponds to randomIndex
+    const randomCard = cards[randomIndex];
+    // Select the card that corresponds to currentIndex
+    const currentCard = cards[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cards[currentIndex] = randomCard;
+    cards[randomIndex] = currentCard;
+  }
+  // Return the shuffled deck
+  return cards;
+};
+let userName;
+let resetBtn;
+let countElement;
+const initGame = () => {
+  // create reset button
+  const resetButtonDiv = document.createElement('div');
+  resetBtn = document.createElement('button');
+  resetBtn.innerHTML = 'reset';
+  document.body.appendChild(resetButtonDiv).appendChild(resetBtn);
+  resetBtn.addEventListener('click', reset);
+  // create count element
+  countElement = document.createElement('h3');
+  countElement.innerHTML = `current score =  ${count}`;
+  document.body.appendChild(resetButtonDiv).appendChild(countElement);
+  // create name
+  const userNameDiv = document.createElement('div');
+  userName = document.createElement('h3');
+  document.body.appendChild(userNameDiv).appendChild(userName);
+  // create input box
+  const inputBoxDiv = document.createElement('div');
+  inputBox = document.createElement('input');
+  submitBtn = document.createElement('button');
+  const name = document.createElement('span');
+  name.innerHTML = 'Input Name :';
+  submitBtn.innerHTML = 'submit';
+  document.body.appendChild(inputBoxDiv).appendChild(name);
+  document.body.appendChild(inputBoxDiv).appendChild(inputBox);
+  document.body.appendChild(inputBoxDiv).appendChild(submitBtn);
+  submitBtn.addEventListener('click', onclick);
+  // create this special deck by getting the doubled cards and
+  // making a smaller array that is ( boardSize squared ) number of cards
+  const doubleDeck = makeDeck();
+  const deckSubset = doubleDeck.slice(0, boardSize * boardSize);
+  deck = shuffleCards(deckSubset);
+
+  // deal the cards out to the board data structure
+  for (let i = 0; i < boardSize; i += 1) {
+    board.push([]);
+    for (let j = 0; j < boardSize; j += 1) {
+      const currentCard = deck.pop();
+      board[i].push(currentCard);
+    }
+  }
+
+  const boardEl = buildBoardElements(board);
+
+  document.body.appendChild(boardEl);
+  // set timer element
+  timer = document.createElement('div');
+  timer.className = 'timer';
+  document.body.appendChild(timer);
+  // set state element
+  state = document.createElement('div');
+  state.className = 'currentState';
+  document.body.appendChild(state);
+};
+
+const timerCountDown = (duration, display) => {
+  let timer = duration; let minutes; let
```

Comment:
Formatting is incorrect here. Please follow the instructions [here](https://bootcamp.rocketacademy.co/course-logistics/required-hardware-and-software#prettier) to download prettier and eslint to format the code and detect syntax errors.